### PR TITLE
Add context to long running operations

### PR DESCRIPTION
**Pull request by MartinForReal**
Add context to long running operations
part of #771

### CODE REVIEW

File: pkg/exporters/k8sexporter/condition/manager.go [hunk 1]
```diff
@@ -25,7 +26,7 @@ import (
 	"k8s.io/node-problem-detector/pkg/types"
 	problemutil "k8s.io/node-problem-detector/pkg/util"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (mmiranda96) [inline pkg/exporters/k8sexporter/condition/manager.go:29]: No need for this change.

**Review by mmiranda96 — CHANGES_REQUESTED** [review]
A couple of comments.

File: pkg/exporters/k8sexporter/condition/manager.go (hunk 1 above)
Comment (MartinForReal) [reply pkg/exporters/k8sexporter/condition/manager.go]: the change is reverted. Thanks!

File: cmd/nodeproblemdetector/node_problem_detector_test.go [hunk 2]
```diff
@@ -81,11 +81,9 @@ func TestNPDMain(t *testing.T) {
 	npdo, cleanup := setupNPD(t)
 	defer cleanup()
 
-	termCh := make(chan error, 2)
-	termCh <- errors.New("close")
-	defer close(termCh)
-
-	if err := npdMain(npdo, termCh); err != nil {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	cancelFunc()
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply cmd/nodeproblemdetector/node_problem_detector_test.go]: fixed the code. Thanks!

**Comment by mmiranda96** [comment]
Thanks for the change!
/lgtm

File: cmd/nodeproblemdetector/node_problem_detector_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (MartinForReal) [reply cmd/nodeproblemdetector/node_problem_detector_test.go]: It turned out that the original code will put error into channel to simulate timeout. And will cal cancel function to do the same thing.

[the remark this answers is not in the document]
Comment (MartinForReal) [reply cmd/nodeproblemdetector/node_problem_detector_test.go]: @vteratipally Could you please take a look?

**Comment by mmiranda96** [comment]
Same as `pull-npd-e2e-test`, it's also caused by the recent migration to EKS.

File: cmd/nodeproblemdetector/node_problem_detector_test.go [hunk 3]
```diff
@@ -79,13 +79,11 @@ func (ne *nullExporter) ExportProblems(*types.Status) {
 
 func TestNPDMain(t *testing.T) {
 	npdo, cleanup := setupNPD(t)
-	defer cleanup()
+	cleanup()
```
Comment (mmiranda96) [inline cmd/nodeproblemdetector/node_problem_detector_test.go:82]: Missing a defer here.

**Review by mmiranda96 — CHANGES_REQUESTED** [review]

File: cmd/nodeproblemdetector/node_problem_detector_test.go (hunk 3 above)
Comment (MartinForReal) [reply cmd/nodeproblemdetector/node_problem_detector_test.go:82]: It turned out that the original code will put error into channel to simulate timeout. And will cal cancel function to do the same thing.

**Comment by mmiranda96** [comment]
e2e-node and e2e-test jobs should pass now.

/retest

File: cmd/nodeproblemdetector/node_problem_detector_test.go (hunk 3 above)
Comment (MartinForReal) [reply cmd/nodeproblemdetector/node_problem_detector_test.go]: fixed.Thanks!

**Comment by MartinForReal** [comment]
<html>
<body>


/test pull-npd-e2e-node
--


<br class="Apple-interchange-newline">
</body>
</html>

**Comment by mmiranda96** [comment]
Issue has been fixed, this should pass now.

/retest-required

**Comment by mmiranda96** [comment]
@vteratipally could you help me approve this PR? Thanks!

**Comment by MartinForReal** [comment]
@vteratipally Could you please take a look? Thanks!

**Comment by mmiranda96** [comment]
Apologies for the delay.

/lgtm

**Review by vteratipally — APPROVED** [review]

**Pull request merged**